api/controllers/kubernetes/install/statemachine.go

```diff
@@ -15,9 +15,16 @@ var validStateTransitions = map[statemachine.State][]statemachine.State{
 	states.StateInstallationConfigurationFailed: {states.StateApplicationConfiguring, states.StateInstallationConfiguring},
 	states.StateInstallationConfigured:          {states.StateApplicationConfiguring, states.StateInstallationConfiguring, states.StateInfrastructureInstalling},
 	states.StateInfrastructureInstalling:        {states.StateSucceeded, states.StateInfrastructureInstallFailed},
+	// TODO: only allow running preflights after infra is installed and before installing the app once the app installation is decoupled from infra installation
+	states.StateAppPreflightsRunning:         {states.StateAppPreflightsSucceeded, states.StateAppPreflightsFailed, states.StateAppPreflightsExecutionFailed},
+	states.StateAppPreflightsExecutionFailed: {states.StateAppPreflightsRunning},
+	states.StateAppPreflightsFailed:          {states.StateAppPreflightsRunning, states.StateAppPreflightsFailedBypassed},
+	states.StateAppPreflightsSucceeded:       {states.StateAppPreflightsRunning},
```

Comment:
why would i want to run again after i preflights have passed? how can i proceed?

Reply:
discussed irl. this will happen in a follow up pr